src/external/vector2.js

```diff
@@ -85,6 +85,14 @@
 		return this;
 	};
 
+	// https://github.com/stackgl/gl-vec2/blob/master/scaleAndAdd.js
+	Vector2.prototype.scaleAndAdd = function(vector, scale) {
```

Comment:
the name `scaleAndAdd` imo suggests that the vector itself gets scaled, then some other `Vector2` is added. Maybe `addScaled` would be a better name.

Reply:
To stay in existing naming convention, `weightedAdd` should be the best solution in context of the existing `weightedAddSelf`.

Reply:
`scale` should be named `scalar` with that in mind.

Reply:
renamed